pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1066,11 +1066,21 @@ func (c *Cluster) saveMonConfig() error {
 		return errors.Wrap(err, "failed to format csi config")
 	}
 
+	maxMonID, err := c.getStoredMaxMonID()
+	if err != nil {
+		return errors.Wrap(err, "failed to save maxMonID")
+	}
+
 	configMap.Data = map[string]string{
 		EndpointDataKey: FlattenMonEndpoints(c.ClusterInfo.Monitors),
-		MaxMonIDKey:     strconv.Itoa(c.maxMonID),
-		MappingKey:      string(monMapping),
-		csi.ConfigKey:   csiConfigValue,
+		// persist the maxMonID that was previously stored in the configmap. We are likely saving info
+		// about scheduling of the mons, but we only want to update the maxMonID once a new mon has
+		// actually been started. If the operator is restarted or the reconcile is otherwise restarted,
+		// we want to calculate the mon scheduling next time based on the committed maxMonID, rather
+		// than only a mon scheduling, which may not have completed.
```

Comment:
I want to check my understanding. If Rook starts a new mon (let's say mon.d to replace mon.b). It starts the deployment for mon.d successfully. But Rook crashes before committing the maxMonID. On the next reconcile, Rook will just attempt to create a new mon.d right, so there is no issue with mon.d existing but the maxMonID being missing?

Reply:
In that case, if mon-d started successfully, but the maxMonId is not committed and the operator restarts, the operator will find mon-d and reset maxMonId to include mon-d. That logic is in the `loadMonConfig()` method around line 246, just below the first change in this PR.

Reply:
Okay, the logic already exists for that. 👍